network/crl.go

```diff
@@ -0,0 +1,370 @@
+/*
+Copyright 2020, Cossack Labs Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/groupcache/lru"
+)
+
+// Errors returned by CRL verifier
+var (
+	ErrInvalidConfigCRLFromCert    = errors.New("Invalid `tls_crl_from_cert` value")
+	ErrInvalidConfigCRLCacheTime   = errors.New("Invalid `tls_crl_cache_time` value")
+	ErrHTTPServerReturnedError     = errors.New("Server returned non-OK status")
+	ErrFetchCRLUnsupportedProtocol = errors.New("Cannot fetch CRL, unsupported protocol")
+	ErrCacheKeyNotFound            = errors.New("Cannot find cached CRL with given URI")
+	ErrOutdatedCRL                 = errors.New("Cannot find cached CRL with given URI")
+)
+
+// --tls_crl_from_cert=<use|ignore>
+const (
+	// If certificate contains CRL distribution point(s), use them
+	crlFromCertUseStr = "use"
```

Comment:
It doesn't seem like we are using constants to generate CLI flags description (from what I see in server and connector).
Isn't it better to agree about flag names, values and description and just hardcode possible values in place where many `flag.{String|Int|Bool}` are created?
If we want to use these constants in tests, they are already accessible (because of the same mod).

Reply:
they will not accessible from `package main` packages which are `acra-connector`/`acra-server`. Or we should declare them in sub-packages

Reply:
These strings constants, possible values for `--tls_*_from_cert` and `--tls_ocsp_required` are now public.
`flag` description will be generated based on them.